python/ray/_private/usage/usage_lib.py

```diff
@@ -277,6 +279,40 @@ def _put_extra_usage_tag(key: str, value: str):
         logger.debug(f"Failed to put extra usage tag, {e}")
 
 
+class _ThirdPartyLibraryUsageFinder(MetaPathFinder):
+    """Custom MetaPathFinder that logs the whitelisted 3rd party library usage."""
+
+    def find_spec(self, fullname, _, target=None):
+        library_name = self._get_library_name(fullname)
+        if (
+            library_name
+            and library_name in usage_constant.THIRD_PARTY_LIBARY_LIBRARIES_WHITELIST
+        ):
+            record_library_usage(library_name)
```

Comment:
Hmm I think this could be pretty expensive because we will have blocking internal kv put for each library for all workers (we can have up to 7 blocking RPCs per worker on startup). 

I actually wonder what if we only do this from driver not worker (is it too sketchy to touch sys.meta_path of driver?)?